### ISSUE TITLE
Postgres Support

### ISSUE THREAD

**Issue by shishkin** [nene/sql-parser-cst#40]
I'm looking forward to this project to support Postgres dialect, so it can unblock [Postgres support in the prettier plugin](https://github.com/nene/prettier-plugin-sql-cst/issues/4).

**Comment by karlhorky** [comment]
Would also love to see PostgreSQL support here 🙌  @nene is this something on the mid-term roadmap?

**Comment by nene** [comment]
Well, it is sort of on the roadmap since starting working on this whole library. But Postgres being pretty big and complex, just the thought of the task ahead puts me off from starting to implement support for it.

**Comment by karlhorky** [comment]
Understood, thanks for the reply. Really appreciate the work that you're doing on both this project and [`sql-formatter`](https://github.com/sql-formatter-org/sql-formatter).

I guess there's also not a way to build on existing parsers / existing schema data or use code generation somehow to reduce the implementation complexity here... 🤔

**Comment by nene** [comment]
> To fully support postgres, it would be good to support psql meta commands (starting with \) and variables (with :), too.

Yep. This is in the plans. But for now I'm concentrating on finishing the support for the actual SQL part.

Another thing to fully support Postgres is having support for PL/pgSQL.

I think both can be implemented as supersets of the base PostgreSQL grammar.
